tasks/upgrade.yml

```diff
@@ -44,6 +44,20 @@
     steps:
       - description: "Upgrading host: {{ item.name }}"
 
+- name: Gather self-heal facts about all gluster hosts in the cluster
+  gluster_heal_facts:
+    name: "{{ volume_item }}"
+    status_filter: self-heal
+  register: self_heal_status
+  until: "self_heal_status | ovirtselfheal == False"
+  delay: 1800
+  retries: 1
```

Comment:
is one retry enough?

Reply:
Yeah i added as per discussion retry for 30 minutes.

Reply:
Maybe use multiple retries with shorted delay wdyt?